File: mcp/floci_backend/api/eks_routes.py
from fastapi import APIRouter
from typing import Dict, Any

from floci_backend.application.eks_service import EksService

def create_eks_router(eks_service: EksService) -> APIRouter:
    router = APIRouter()

    @router.get('/eks/overview')
    async def get_overview() -> Dict[str, Any]:
        return await eks_service.get_overview()

    return router

File: sidecar/src/routes/eksRoutes.ts
import { Router } from 'express';
import { EksService } from '../application/eksService';

export const createEksRouter = (eksService: EksService) => {
  const router = Router();

  router.get('/eks/overview', async (_request, response) => {
    try {
      response.json(await eksService.getOverview());
    } catch (error: any) {
      response.status(500).json({ error: error.message || 'Failed to read EKS overview' });
    }
  });

  return router;
};
